shell/platform/windows/flutter_windows_view.cc

```diff
@@ -50,6 +50,9 @@ FlutterWindowsView::FlutterWindowsView(
 
 FlutterWindowsView::~FlutterWindowsView() {
   DestroyRenderSurface();
```

Comment:
This is fine but I think it'd be preferable if the view was deregistered from the engine before its state is torn down. This way the engine does not point to a "bad" view.

Reply:
What are you considering when the state is torn down? At the start of the destructor being called?

Reply:
I meant that I would deregister the view before destroying the render surface. So yup, deregistering the view at the start of the destructor would work